ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/VersusBuyAndHoldCriterion.java

```diff
@@ -42,26 +43,26 @@ public VersusBuyAndHoldCriterion(AnalysisCriterion criterion) {
     }
 
     @Override
-    public double calculate(TimeSeries series, TradingRecord tradingRecord) {
+    public Num calculate(TimeSeries series, TradingRecord tradingRecord) {
         TradingRecord fakeRecord = new BaseTradingRecord();
         fakeRecord.enter(series.getBeginIndex());
         fakeRecord.exit(series.getEndIndex());
 
-        return criterion.calculate(series, tradingRecord) / criterion.calculate(series, fakeRecord);
+        return criterion.calculate(series, tradingRecord).dividedBy(criterion.calculate(series, fakeRecord));
     }
 
     @Override
-    public double calculate(TimeSeries series, Trade trade) {
+    public Num calculate(TimeSeries series, Trade trade) {
         TradingRecord fakeRecord = new BaseTradingRecord();
         fakeRecord.enter(series.getBeginIndex());
         fakeRecord.exit(series.getEndIndex());
 
-        return criterion.calculate(series, trade) / criterion.calculate(series, fakeRecord);
+        return criterion.calculate(series, trade).dividedBy(criterion.calculate(series, fakeRecord));
     }
 
     @Override
-    public boolean betterThan(double criterionValue1, double criterionValue2) {
-        return criterionValue1 > criterionValue2;
+    public boolean betterThan(Num criterionValue1, Num criterionValue2) {
+        return criterionValue1.isGreaterThan(criterionValue2);
     }
 
     @Override
```

Comment:
VersusBuyAndHold looks good.  Can probably remove `toString()`.  Would be a tiny change so I'm fine with making that change in this PR.